src/main/java/racing/domain/Cars.java

```diff
@@ -26,11 +36,29 @@ Cars move(Movable movable) {
         return new Cars(carList);
     }
 
-    private List<Car> initCars(int unit) {
-        List<Car> cars = new ArrayList<>(unit);
-        for (int i = 0; i < unit; i++) {
-            cars.add(i, new Car());
+    List<Car> getCarsWithMaxDistance(final Cars snapshot) {
+        List<Car> winningCars = new ArrayList<>();
+        int maxDistance = getMaxDistance(snapshot);
+
+        for (Car car : snapshot.getCars()) {
+            addCarLocatedAtMaxDistance(winningCars, car, maxDistance);
         }
-        return cars;
+
+        return winningCars;
+    }
+
+    private void addCarLocatedAtMaxDistance(List<Car> winningCars, Car car, int maxDistance) {
+        if (car.getDistance() == maxDistance) {
```

Comment:
`Car`에서 값을 꺼내는 대신, 메시지를 보내서 처리해보면 어떨까요?
```suggestion
        if (car.isLocatedAt(maxDistance)) {

```